File: Sources/Scratchpad/Drawing/Models/StrokePoint.swift
import CoreGraphics

/// A `Codable` stand-in for `CGPoint`, which does not itself conform to
/// `Codable`. Kept as plain `Double` fields so drawings persist in a format
/// with no Apple-specific types, per the project's Windows-portability goal.
struct StrokePoint: Codable, Hashable {
    var x: Double
    var y: Double

    init(x: Double, y: Double) {
        self.x = x
        self.y = y
    }

    init(_ point: CGPoint) {
        self.x = point.x
        self.y = point.y
    }

    var cgPoint: CGPoint {
        CGPoint(x: x, y: y)
    }
}

File: Windows/Scratchpad/Drawing/StrokePoint.cs
namespace Scratchpad.Drawing;

/// A plain point, kept independent of any UI-framework point type so
/// drawings persist in a format with no WPF-specific types.
public readonly record struct StrokePoint(double X, double Y)
{
    public static StrokePoint FromWindowsPoint(System.Windows.Point point) => new(point.X, point.Y);

    public System.Windows.Point ToWindowsPoint() => new(X, Y);
}
